### DISCUSSION TITLE
How to I move to  column with given name - and back?

### DISCUSSION THREAD

**Discussion by halloleo** [saulpw/visidata#1582] [Q&A] [answered]
In a CSV with many columns how can I jump to  a column given its header name? 

For example, I have a CSV file were the second column is called "**type**" and the 25th column is called "**distance**" and for a few rows I want to edit the "**distance**" values manually on the grounds of the "**type**" details and then save the CSV file back.

So when I am in a given row I need to jump between between these two columns back and forth -- what is the easiest way to do this?

**Comment by frosencrantz** [comment, marked as answer]
Hi @halloleo,
There are several options.  I have a solution that is close to what you want, but I think I would go about it differently, and reposition the columns closer to each other so you can see them both on the screen at once. 
- Use the `c` `go-col-regex` command and as input give a regex that matches the two column names `^(colname1|colname2)$`.  Subsequent times you hit `c`, you can hit `Enter` and it reuses the last value, and so it should jump to the other column.  This does what you asked.
- Make one or both column `key-col` `!`.  The key columns are anchored on the left side of the screen.  This has some significance for other commands like generating a frequency sheet with multiple columns `gF` `freq-keys`, and for joins.
- Reorder the columns so they are next to each other.  Use the `slide-` commands to reposition the columns:
   - `Shift+H` / `slide-left` : slide current column left
   - `zShift+H` / `slide-left-n` : slide current column N positions to the left
   - `gShift+H` / `slide-leftmost` : slide current column all the way to the left of sheet
   - `Shift+L` / `slide-right` : slide current column right
   - `zShift+L` / `slide-right-n` : slide current column N positions to the right
   - `gShift+L` / `slide-rightmost` : slide current column all the way to the right of sheet

**Reply by halloleo** [reply]
Much appreciated @frosencrantz! The first solution looks the best to me. 😄

Regarding the key column solution: Does making a column a key column reorder the underlying file on save? That's what I totally want to avoid!

**Reply by geekscrapy** [reply]
It won't change anything until you save it again. When you save, the columns will be in the order you see them (WYSIWYG), i.e. key columns will be first

**Reply by halloleo** [reply]
Well, that's exactly it what I want to avoid, because I have to save the edited file at the end.

BTW, I've just tried the `c` command -- and it is ***wonderful*** !

**Reply by frosencrantz** [reply]
@halloleo glad to hear that `c` command works well. 

I think if you toggle off the key column before saving the file, the column order will not be changed.

**Reply by halloleo** [reply]
Ta. Will try this.
